### PR TITLE
Alex/#723#711

### PR DESCRIPTION
**Pull request by AlexandruPopovici**
- closes https://github.com/specklesystems/speckle-server/issues/723 -> Example stream https://speckle.xyz/streams/99abc74dd4/commits/b32fdcf171 
- closes https://github.com/specklesystems/speckle-server/issues/711 -> Example streams https://latest.speckle.dev/streams/0c6ad366c4/commits/912d83412e ; https://speckle.xyz/streams/99abc74dd4/commits/b32fdcf171
- closes https://github.com/specklesystems/speckle-server/issues/708 -. Example stream https://latest.speckle.dev/streams/0c6ad366c4/commits/462dabba78 (this one is a bit strange because there are two display values with unique ids inside the hierarchy, so they are both converted. But since they have identical geometry, they double up and cause z-fighting. Not sure if this is indented behaviour)
- closes https://github.com/specklesystems/speckle-server/issues/282 too :)

### CODE REVIEW

**Comment by didimitrie** [comment]
> this one is a bit strange because there are two display values with unique ids inside the hierarchy, so they are both converted. But since they have identical geometry, they double up and cause z-fighting. Not sure if this is indented behaviour)

In that specific case, it is intended behaviour. @tsvilans / @izzylys doubled up some data without needing to (ie the brep value already contains a mesh, but we also have the mesh as direct display value. It might be that the data structure is a direct result of how we were (not) handling things so far. 

We could return from the conversion if we encounter a `displayValue`, but this opens up another can of worms.

File: packages/viewer/src/modules/converter/Converter.js [hunk 1]
```diff
@@ -186,7 +202,19 @@ export default class Coverter {
       const type = this.getSpeckleType(obj)
       if (this[`${type}ToBufferGeometry`]) {
         return await this[`${type}ToBufferGeometry`](obj.data || obj, scale)
-      } else return null
+      }
+      /**
+       * Regarding #723. This would be more generic and possibly handle other
+       * types with missing direct convertor, however I don't feel it is the
+       * 'convert' fuction's place to handle this...
+       */
```
Comment (didimitrie) [inline packages/viewer/src/modules/converter/Converter.js:210]: Agreed. The whole traversal & conversion logic needs to be cleaned up massively anyway...

File: packages/viewer/src/modules/converter/Converter.js [hunk 2]
```diff
@@ -620,6 +725,9 @@ export default class Coverter {
       this.PointToVector3(plane.normal).normalize()
     )
     m.setPosition(this.PointToVector3(plane.origin))
+    /**
+     * I think scaling should be done first.
+     */
```
Comment (didimitrie) [inline packages/viewer/src/modules/converter/Converter.js:730]: You know you're the boss, so next pass over this one, feel free :)

**Review by didimitrie — APPROVED** [review]

File: packages/viewer/src/modules/converter/Converter.js [hunk 3]
```diff
@@ -173,6 +176,19 @@ export default class Coverter {
     this.activePromises -= childrenConversionPromisses.length
   }
 
+  directConverterExists(obj) {
+    return this[`${this.getSpeckleType(obj)}ToBufferGeometry`] !== undefined
+  }
+
+  getDisplayValue(obj) {
```
Comment (clairekuang) [inline packages/viewer/src/modules/converter/Converter.js:183]: So `displayMesh` was our really old version of fallback geometry values: I think the viewer ought to render only what's inside `displayValue` if it exists, and only to check for `displayMesh` if not (to maintain backwards compatibility)

**Pull request merged**